iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientOptions.java

```diff
@@ -138,10 +138,10 @@
     private final int receiveInterval = RECEIVE_PERIOD_MILLIS;
 
     /**
-     * This option specifies the type of gateway to which the device/module client is connecting while
-     * specifying the property of GatewayHostName. By default, the value is EDGE.
+     * This option specifies the type of gateway to which the device/module client is connecting. By default,
+     * the value is None.
      */
     @Getter
     @Builder.Default
-    private final GatewayType gatewayType = GatewayType.EDGE;
+    private final GatewayType gatewayType = GatewayType.None;
```

Comment:
Sorry to waffle on this, but I realize that we probably can't have "None" be the default since existing edge users would be broken since they now need to pass in this new option. To avoid breaking changes, we'll need just the two gateway types here and the default is Edge.

We'll just need to mention in the javadocs that this field is ignored if no GatewayHostname is provided

Reply:
Synced up offline. Ideally, the value of `GatewayHostName` should be ignored if the `GatewayType` is set to "None" (by default or specified explicitly), thus the existing users would have to pass in a `ClientOption` with "Edge" `GatewayType` while connecting to Edge, which is a breaking change for them.

Therefore, changing this back to 2 enum values and adding additional comment to explain this.